indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -500,6 +502,7 @@ public void handle()
 
           boolean allocationSuccess = changeTaskCount(desiredTaskCount);
```

Comment:
Would it make sense to move the metric emission of desired task count to line 505 so that the desired task count is only reported when changing task count is successful? The check on desiredTaskCount > 0 looks similar to the check in the `chageTaskCount` method

Reply:
I think we want the emit the desired task count even when we are not successful in scaling.

Reply:
In my view, that metric has nothing to do with the supervisor and its action but just about the autoscaler and its recommendations. we can exit early if count is -1, lmk if you want me to make that change instead.